src/java.base/share/classes/java/math/MutableBigInteger.java

```diff
@@ -528,7 +547,7 @@ public String toString() {
      * Like {@link #rightShift(int)} but {@code n} can be greater than the length of the number.
      */
     void safeRightShift(int n) {
-        if (n/32 >= intLen) {
+        if (n >= bitLength()) {
```

Comment:
The commit message for this reads `More accurate condition for MBI.safeRightShift()`.
If the old version works, please switch back. But if this is a genuine bug, then it needs a separate bug issue and PR.

Reply:
I tend to agree with your point of view, but
- it is a bug: then it needs to be tracked and improved in a separate JBS issue and a separate PR
- or it is not a bug, then it should not be modified in the scope of this PR if it works in the original version
You can, of course, add a comment about the precondition you identified by working on this PR, as you did in other methods ("Assumes that ...).